sources/iTermScrollAccumulator.m

```diff
@@ -92,33 +76,18 @@ - (CGFloat)accumulatedDeltaYForTrackpadEvent:(NSEvent *)event {
     _accumulatedDeltaY += delta;
     const CGFloat absAccumulatedDelta = fabs(_accumulatedDeltaY);
     int roundDelta;
-
-    if (absAccumulatedDelta > 0.5) {
-        // The rounded accumulated value will not be zero so consume it.
-        roundDelta = round(_accumulatedDeltaY);
+    
+    // Deltas will be accumulated into _accumulatedDeltaY.
+    // If it is large (>1), return its integer part. This enables quick scroll, which feels like natural trackpad.
+    // If `delta * _accumulatedDeltaY < 0`, it means turnaround. Round delta to turn around quickly (fabs 0.5 is enough to move).
+    // If it is not large enough, return 0 and keep accumulating.
+    if (absAccumulatedDelta > 1) {
+        roundDelta = _accumulatedDeltaY > 0 ? floor(_accumulatedDeltaY) : ceil(_accumulatedDeltaY);
```

Comment:
There's a `RoundTowardZero` function on line 109 that does just want you want here.